test/test/alloc/AllocTests.java

```diff
@@ -55,4 +61,69 @@ public void humongous(TestProcess p) throws Exception {
         assert out.contains("java/io/ByteArrayOutputStream.toByteArray;");
         assert out.contains("G1CollectedHeap::humongous_obj_allocate");
     }
+
+    // Without liveness tracking, results won't change except for the sampling
+    // error.
+    @Test(mainClass = RandomBlockRetainer.class, args = "1.0", inputs = { "false" })
+
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.0", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.1", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.2", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.3", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.4", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.5", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.6", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.7", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.8", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.9", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "1.0", inputs = { "true" })
+    public void liveness(TestProcess p) throws Exception {
+        final long TOTAL_BYTES = 50000000;
+        final double tolerance = 0.10;
+
+        final boolean live = Boolean.parseBoolean(p.inputs()[0]);
+        final double keepChance = live ? Double.parseDouble(p.test().args()) : 1.0;
+
+        Output out = p.profile("--alloc 1k --total -o collapsed" + (live ? " --live" : ""));
```

Comment:
`--live` is supported on JDK 11+ only. The does should not run or should fail on JDK 8.